pkg/reconciler/instances/istio/action.go

```diff
@@ -225,8 +235,10 @@ func canUpdate(ver actions.IstioVersion, logger *zap.SugaredLogger) bool {
 	dataPlaneVsTarget := targetHelperVersion.compare(&dataPlaneHelperVersion)
 
 	if pilotVsTarget == -1 || dataPlaneVsTarget == -1 {
-		logger.Errorf("Downgrade detected from pilot: %s and data plane: %s to version: %s - finishing...", ver.PilotVersion, ver.DataPlaneVersion, ver.TargetVersion)
-		return false
+		if !targetHelperVersion.isDowngradePermitted(&pilotHelperVersion) || !targetHelperVersion.isDowngradePermitted(&dataPlaneHelperVersion) {
+			logger.Errorf("Downgrade detected from pilot: %s and data plane: %s to version: %s - finishing...", ver.PilotVersion, ver.DataPlaneVersion, ver.TargetVersion)
+			return false
+		}
 	}
```

Comment:
We are mixing here the upgrade and the downgrade case, what might be confusing for one who reads that. The `canUpdate` function says now true if either update or downgrade is allowed. 
What about creating a new function like "canReconcile" which either returns true if update or downgrade is possible?

Reply:
As discussed, "canReconcile" could indicate an install as well. Hence I've added an additional log to clearly specify the valid downgrade scenario - which is also an "update".